src/dimension.c

```diff
@@ -564,13 +588,10 @@ hyperspace_calculate_point(Hyperspace *hs, HeapTuple tuple, TupleDesc tupdesc)
 								NameStr(d->fd.column_name)),
 						 errhint("Columns used for time partitioning cannot be NULL")));
 
-			p->coordinates[p->num_coords++] = time_value_to_internal(datum, d->fd.column_type, false);
+			p->coordinates[p->num_coords++] = time_value_to_internal(datum, dimtype, false);
 		}
 		else
-		{
-			p->coordinates[p->num_coords++] =
-				partitioning_func_apply_tuple(d->partitioning, tuple, tupdesc);
-		}
+			p->coordinates[p->num_coords++] = DatumGetInt32(datum);
```

Comment:
Should we switch the condition to `IS_ClOSED_DIMENSION` and have this case first? It's currently very easy to skip since it's so short and indented the same as the last block

Reply:
I'd actually rather do a `switch` then, which also protects against `DIMENSION_TYPE_ANY`